optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverManagerConfig.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.config.solver;
+
+import java.util.concurrent.ThreadFactory;
+
+import org.optaplanner.core.config.AbstractConfig;
+import org.optaplanner.core.config.util.ConfigUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolverManagerConfig extends AbstractConfig<SolverManagerConfig> {
+
+    public static final String PARALLEL_SOLVER_COUNT_AUTO = "AUTO";
+
+    private static final Logger logger = LoggerFactory.getLogger(SolverManagerConfig.class);
+
+    protected SolverConfig solverConfig;
+    protected String parallelSolverCount = null;
+    protected Class<? extends ThreadFactory> threadFactoryClass = null;
+
+    // Future features:
+    // throttlingDelay
+    // congestionStrategy
+
+    // ************************************************************************
+    // Constructors and simple getters/setters
+    // ************************************************************************
+
+    public SolverManagerConfig() {
+    }
+
+    public SolverManagerConfig(SolverConfig solverConfig) {
+        this.solverConfig = solverConfig;
+    }
+
+    public SolverConfig getSolverConfig() {
+        return solverConfig;
+    }
+
+    public void setSolverConfig(SolverConfig solverConfig) {
+        this.solverConfig = solverConfig;
+    }
+
+    public String getParallelSolverCount() {
+        return parallelSolverCount;
+    }
+
+    public void setParallelSolverCount(String parallelSolverCount) {
+        this.parallelSolverCount = parallelSolverCount;
+    }
+
+    public Class<? extends ThreadFactory> getThreadFactoryClass() {
+        return threadFactoryClass;
+    }
+
+    public void setThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryClass) {
+        this.threadFactoryClass = threadFactoryClass;
+    }
+
+    // ************************************************************************
+    // With methods
+    // ************************************************************************
+
+    public SolverManagerConfig withParallelSolverCount(String parallelSolverCount) {
+        this.parallelSolverCount = parallelSolverCount;
+        return this;
+    }
+
+    public SolverManagerConfig withThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryClass) {
+        this.threadFactoryClass = threadFactoryClass;
+        return this;
+    }
+
+    // ************************************************************************
+    // Builder methods
+    // ************************************************************************
+
+    public Integer resolveParallelSolverCount() {
+        int availableProcessorCount = getAvailableProcessors();
+        Integer resolvedParallelSolverCount;
+        if (parallelSolverCount == null || parallelSolverCount.equals(PARALLEL_SOLVER_COUNT_AUTO)) {
+            // Leave one for the Operating System and 1 for the solver thread, take the rest
```

Comment:
Not sure why leave just 1 for solver thread. Shouldn't solver threads get most of the CPU unless the user wants otherwise?

Reply:
This comment is actually false, so I am removing it.
By default, we only take half of the CPU threads. The reason that taking too many introduces freezes (including on the consumer threads, the REST threads, the OS, etc). That gives a very poor user experience. People don't like it. If they want to live closer to the border, they can configure a number manually (or use the availableProcesses variable).